src/go/rpk/pkg/config/weak.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2020 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package config
+
+import (
+	"fmt"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+// This file contains weak params type, including basic types support (bool,
+// int, and string) and one_or_many support for different types.
+//
+// The use of this file is to support our transition to a strongly typed
+// config file and our migration away from cobra and mapstructure.
+
+// weakBool is an intermediary boolean type to be used during our transition
+// to strictly typed configuration parameters. This will allow us to support
+// weakly typed parsing:
+//
+//   - int to bool (true if value != 0)
+//   - string to bool (accepts: 1, t, T, TRUE, true, True, 0, f, F, FALSE,
+//     false, False. Anything else is an error)
+type weakBool bool
+
+func (wb *weakBool) UnmarshalYAML(n *yaml.Node) error {
+	switch n.Tag {
+	case "!!bool":
+		bValue, err := strconv.ParseBool(n.Value)
```

Comment:
I'd just name this `b`.

Also, why does this case and the `!!string` case both use `ParseBool`? Is this just a yaml thing, where `true` or `false` is always `!!bool`?

Reply:
Yes, true or false is always `!!bool` and `n.Value` is a string so we can use `ParseBool` for both cases